server/odc-service/src/main/java/com/oceanbase/odc/service/db/util/OBMysqlCallFunctionCallBack.java

```
@@ -76,23 +81,39 @@ public CallFunctionResp doInConnection(Connection con) throws SQLException, Data
             if (this.timeoutSeconds > 0) {
                 stmt.setQueryTimeout(this.timeoutSeconds);
             }
-            PLOutParam plOutParam = new PLOutParam();
             try (ResultSet res = stmt.executeQuery(sqlBuilder.toString())) {
                 if (!res.next()) {
-                    plOutParam.setValue(function.getReturnValue());
+                    return generateDefaultReturnValue();
+                }
+                JdbcQueryResult jdbcQueryResult = new JdbcQueryResult(res.getMetaData(), rowDataMapper);
+                jdbcQueryResult.addLine(res);
+                List<List<Object>> rows = jdbcQueryResult.getRows();
+                if (CollectionUtils.size(rows) == 1 && Objects.nonNull(rows.get(0))
+                        && CollectionUtils.size(rows.get(0)) == 1) {
+                    CallFunctionResp callFunctionResp = new CallFunctionResp();
+                    PLOutParam plOutParam = new PLOutParam();
+                    plOutParam.setValue(String.valueOf(rows.get(0).get(0)));
                     plOutParam.setDataType(function.getReturnType());
                     callFunctionResp.setReturnValue(plOutParam);
+                    callFunctionResp.setOutParams(null);
                     return callFunctionResp;
+                } else {
```

The behavior changed if the resultset's size was not 1. Is it expected?

---

Not expected, should throw an exception